core/src/main/java/com/linecorp/armeria/internal/common/AbstractKeepAliveHandler.java

```diff
@@ -194,6 +197,10 @@ public final void onPing() {
             return;
         }
 
+        if (connectionIdleTimeNanos > 0 && keepAliveOnPing) {
+            lastConnectionIdleTime = System.nanoTime();
+        }
```

Comment:
I believe this is what we should do based on the `onReadOrWrite` implementation. Let me know if it doesn't look right or misses some other code path that leads `pings` to be registered.